xbmc/dialogs/GUIDialogOK.cpp

```diff
@@ -72,9 +70,18 @@ void CGUIDialogOK::ShowAndGetInput(CVariant heading, CVariant line0, CVariant li
   dialog->Open();
 }
 
+void CGUIDialogOK::OnInitWindow()
+{
+  SET_CONTROL_HIDDEN(CONTROL_CANCEL_BUTTON);
+  SET_CONTROL_HIDDEN(CONTROL_PROGRESS_BAR);
+  SET_CONTROL_FOCUS(CONTROL_OKAY_BUTTON, 0);
+
```

Comment:
I would invert the responsibility here: instead of every subclass setting unneeded buttons to hidden, `Base` set the entire control start to max range to hidden, and subclasses override for the ones they use.